lib/src/store.rs

```diff
@@ -39,10 +41,14 @@ use crate::tree_builder::TreeBuilder;
 /// Wraps the low-level backend and makes it return more convenient types. Also
 /// adds caching.
 pub struct Store {
+    repo_path: PathBuf,
```

Comment:
To be able to access the right path, I sneaked `repo_path` in here. Probably not the best way to go about it. Possibly it could be all moved into it's own `struct LocalChangesTracker`.

Or maybe local cachnges should go into backend  (see other comment).